src/utils/webrtc/models/CallParticipantModel.js

```diff
@@ -0,0 +1,249 @@
+/**
+ *
+ * @copyright Copyright (c) 2019, Daniel Calviño Sánchez (danxuliu@gmail.com)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+export const ConnectionState = {
+	NEW: 'new',
+	CHECKING: 'checking',
+	CONNECTED: 'connected',
+	COMPLETED: 'completed',
+	DISCONNECTED: 'disconnected',
+	DISCONNECTED_LONG: 'disconnected-long', // Talk specific
+	FAILED: 'failed',
+	FAILED_NO_RESTART: 'failed-no-restart', // Talk specific
+	CLOSED: 'closed',
+}
+
+export default function CallParticipantModel(options) {
```

Comment:
You can use classes in javascript, with constructors and all instead of functions.
This would avoid you playing with prototypes and binding I think :)